src/libtools/graphmemoryvisitor.cc

```diff
@@ -0,0 +1,57 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "../../include/votca/tools/graphmemoryvisitor.h"
+#include "../../include/votca/tools/graph_bf_visitor.h"
+#include "../../include/votca/tools/graphvisitor.h"
+
+#include "../../include/votca/tools/edge.h"
+#include "../../include/votca/tools/graph.h"
+#include "../../include/votca/tools/graphnode.h"
+
+using namespace std;
+
+namespace votca {
+namespace tools {
+
+// Add the distance to the node that has not yet been explored
+void GraphMemoryVisitor::exploreNode(pair<Index, GraphNode>& p_gn, Graph& g,
+                                     const Edge& ed) {
+  // Determine if the node has already been explored
+  Index vertex = p_gn.first;
+  if (vertex == startingVertex_) {
+    p_gn.second.addInt("Dist", 0);
+    // Update the graph with new graph node
+    g.setNode(p_gn);
+  } else {
+    // Node has not been explored
+    if (explored_.count(vertex) == 0) {
+      Index prev_vertex = ed.getOtherEndPoint(vertex);
+      GraphNode gn_prev = g.getNode(prev_vertex);
+      p_gn.second.addInt("Dist", gn_prev.getInt("Dist") + 1);
```

Comment:
you use a value property for the properties of the  graph that is dangerous? What happens if I have a value Dist on my object?